lib/network/struct.go

```diff
@@ -194,15 +194,13 @@ type Entity struct {
 	ID EntityID
 	// A slice of addresses of where that Id might be found
 	Addresses []string
-	// used to return the next available address
-	iter int
 }
 
 // EntityID uniquely identifies an Entity struct
 type EntityID uuid.UUID
 
-// Equals returns true if both EntityID are equal or false otherwise.
-func (eid EntityID) Equals(other EntityID) bool {
+// Equal returns true if both EntityID are equal or false otherwise.
+func (eid EntityID) Equal(other EntityID) bool {
```

Comment:
For other uuids this function is still called `Equals` (like for TreeID). I would prefer to keep named like before. If you want to rename it, open an issue and rename all `Equals` functions.